### DISCUSSION TITLE
RFC7296: ESP encap in UDP on port 4500 with no NAT device

### DISCUSSION THREAD

**Discussion by KonradLopuszynski** [strongswan/strongswan#364] [Q&A]
**Hi!**

I'm trying to configure vti tunnel supporting nat-traversal with no NAT device between.
As it has been stated in the **RFC7296** 2.23.:
> _An initiator can use port 4500 for both IKE and ESP, regardless of **whether or not there is a NAT**, even at the beginning of IKE.  When either side is using port 4500, sending ESP with UDP encapsulation is not required, but **understanding received UDP-encapsulated ESP packets is required**._

and my question is: 
**Is it possible to configure StrongSwan/xfrm in that way to process 'plain' ESP packets and simultaneously ESP encapsulated in UDP on port 4500 using the same CHILD SA. If so, what circumstances must be met to make this possible?**

Great thanks for answer in advance!

Best Regards,
Konrad Lopuszynski

**Comment by Thermi** [comment]
The kernel processes them. Just permit protocol ESP and all traffic to your IKE ports (default UDP 500 and 4500).
AFAIR there's an issue on the wiki's issue tracker that explores the topic of processing plain ESP and UDP encapsulated ESP where the SA is configured for only one of those.

if libipsec is used, plain ESP is not supported (and can't be), so the case is not solvable there.

**Reply by KonradLopuszynski** [reply]
Thanks a lot for the answer!

> AFAIR there's an issue on the wiki's issue tracker that explores the topic of processing plain ESP and UDP encapsulated ESP where the SA is configured for only one of those. 

Could you please point to where is the description and thread for the issue you mentioned? I'm probably testing an identical scenario and I want to make sure that it is the reason that the packets are dropped. I cannot find it on your wiki forum and issue tracker as well.
Thanks again!

Best Regards,
Konrad Lopuszynski

**Comment by Thermi** [comment]
https://wiki.strongswan.org/issues/2670
